File: Valid Parentheses/valid_parentheses.py
class Solution:
    def isValid(self, parentheses_list: str) -> bool:
        str_stack = []
        parentheses_dict = {"}": "{", "]": "[", ")": "("}
        for parentheses in parentheses_list:
            if parentheses in "{[(":
                str_stack.append(parentheses)
            elif (
                len(str_stack) == 0
                or parentheses_dict.get(parentheses) != str_stack.pop()
            ):
                return False

        return len(str_stack) == 0


if __name__ == "__main__":
    Solution().isValid("]]")

File: Valid Parentheses/Valid_Parentheses.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <map>
#include <stack>

using namespace std;

class Solution {
public:
    bool isValid(string s) {
        map<char, char> m { 
            { '(', ')' },
            { '{', '}' },
            { '[', ']' } };
        
        stack<char> st;
        
        for (auto i : s)
        {
            if (i == '(' || i == '{' || i == '[')
                st.push(m[i]);

            if (i == ')' || i == '}' || i == ']')
            {
                if (st.size() == 0) return false;

                if (st.top() == i) 
                    st.pop();
                else 
                    return false;
            }
                
        }

        return st.size() == 0;
    }
};

int main()
{
    Solution s;
    //string str("(){}[]");
    string str("[");
    
    bool b = s.isValid(str);
    return 0;
}